### PR TITLE
PLANNER-2354 Use Quarkus configuration to set daemon mode

### PR DESCRIPTION
**Pull request by yurloc**
### JIRA
https://issues.redhat.com/browse/PLANNER-2354

### Checklist
- [x] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>full downstream build</b> please add comment: <b>Jenkins run fdb</b>
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/SolverBuildTimeConfig.java [hunk 1]
```diff
@@ -33,11 +33,12 @@
...
     /**
-     * Enable runtime assertions to detect common bugs in your implementation during development.
-     * Defaults to {@link EnvironmentMode#REPRODUCIBLE}.
+     * Enable daemon mode. In daemon mode, non-early termination pauses the solver instead of stopping it,
+     * until the next problem fact change arrives. This is often useful for real-time planning.
+     * Defaults to "false".
      */
     @ConfigItem
-    Optional<EnvironmentMode> environmentMode;
+    Optional<Boolean> daemon;
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/main/java/org/optaplanner/quarkus/deployment/SolverBuildTimeConfig.java:41]: In SolverConfig.java, "daemon" is just after "EnviromnentMode". The XSD enforces this order in solverConfig.xml.
Let's keep that order for this file too, and the docs of quarkus properties?

File: optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/OptaPlannerProcessorSolverPropertiesTest.java [hunk 2]
```diff
@@ -42,13 +43,13 @@
...
-            .overrideConfigKey("quarkus.optaplanner.solver.environment-mode",
-                    "FULL_ASSERT")
+            .overrideConfigKey("quarkus.optaplanner.solver.daemon", "true")
+            .overrideConfigKey("quarkus.optaplanner.solver.domain-access-type", "REFLECTION")
+            .overrideConfigKey("quarkus.optaplanner.solver.environment-mode", "FULL_ASSERT")
             .overrideConfigKey("quarkus.optaplanner.solver.move-thread-count", "2")
             .overrideConfigKey("quarkus.optaplanner.solver.termination.spent-limit", "4h")
             .overrideConfigKey("quarkus.optaplanner.solver.termination.unimproved-spent-limit", "5h")
             .overrideConfigKey("quarkus.optaplanner.solver.termination.best-score-limit", "0")
-            .overrideConfigKey("quarkus.optaplanner.solver.domain-access-type", "REFLECTION")
```
Comment (ge0ffrey) [inline optaplanner-quarkus-integration/optaplanner-quarkus/deployment/src/test/java/org/optaplanner/quarkus/OptaPlannerProcessorSolverPropertiesTest.java:51]: Nitpick:
in the SolverConfig.java, domain-access-type is before termination, but after move-thread-count. Can we keep that order here too?

**Review by ge0ffrey — APPROVED** [review]
LGTM, @yurloc Please see two suggestions about order.

**Comment by yurloc** [comment]
Updated all the affected code to follow the canonical order:
- environmentMode
- daemon
- moveThreadCount
- domainAccessType
- termination

**Pull request merged**